### PR TITLE
math functions and high pass filter

### PR DESCRIPTION
**Pull request by EwoudSmeur**
The high pass filter uses doubles to avoid numerical errors.

This is some of the code I use for the Cyclone, planning to merge all of it piece by piece like this.

The 2D matrix stuff I actually don't use for the Cyclone at this point, but I thought it could be useful in the future.

### CODE REVIEW

File: sw/airborne/math/pprz_algebra_float.c [hunk 1]
```diff
@@ -651,10 +755,11 @@ float float_mat_det_4d(float m[16])
...
 void float_mat_inv_4d(float invOut[16], float mat_in[16])
 {
-    float_mat_adjoint_4d(mat_in, invOut);
+  float_mat_adjoint_4d(mat_in, invOut);
 
-    float inv_det = 1.0f / float_mat_det_4d(mat_in);
-    int i;
-    for(i = 0; i < 16; ++i)
-        invOut[i] = invOut[i] * inv_det;
+  float inv_det = 1.0f / float_mat_det_4d(mat_in);
```
Comment (kirkscheper) [inline sw/airborne/math/pprz_algebra_float.c]: check if matrix is invertible here before the divide?

File: sw/airborne/math/pprz_algebra_float.c [hunk 2]
```diff
@@ -609,38 +713,38 @@ float float_mat_det_4d(float m[16]);
```
Comment (kirkscheper) [inline sw/airborne/math/pprz_algebra_float.c]: The lines above this shouldn't be needed, can you remove them plz :) less more is more better...

Also, if you are super generous, can you add function descriptors for some more of these functions (very optional)?

**Comment by EwoudSmeur** [comment]
Thanks Kirk. I copied the check for the 2D case, and if I remember correctly Gautier advised me to use 1e-4 instead of something like FLOAT_EPSILON. I am not sure why, somehow 1e-4 seems a bit arbitrary to me.

**Comment by kirkscheper** [comment]
indeed 1e-4 is arbitrary, I usually use 1e-5 :p 

You can define something in std.h

File: sw/airborne/math/pprz_algebra_float.c (hunk 2 above)
Comment (EwoudSmeur) [reply sw/airborne/math/pprz_algebra_float.c]: Without this I get a warning for a missing prototype...

**Comment by gautierhattenberger** [comment]
looks good

**Pull request merged**